app/demo-geant-integration/HitRootIO.hh

```diff
@@ -0,0 +1,105 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file demo-geant-integration/HitRootIO.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <G4ThreadLocalSingleton.hh>
+#include <G4Types.hh>
+#include <G4VHit.hh>
+
+#include "celeritas_config.h"
+#include "corecel/Assert.hh"
+
+class TFile;
+class TTree;
+class TBranch;
+class G4Event;
+
+namespace demo_geant
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Example of output event structure stored in ROOT.
+ */
+class HitRootEvent
+{
+  public:
+    //!@{
+    //! \name Type aliases
+    using HitContainer = std::map<std::string, std::vector<G4VHit*>>;
+    //!@}
+
+  public:
+    HitRootEvent() = default;
+
+    void SetEventID(int id) { event_id_ = id; }
+    HitContainer* GetHCMap() { return &hcmap_; }
+
+  private:
+    int event_id_{0};
+    HitContainer hcmap_;
+};
```

Comment:
This class doesn't have any advantage over a simple struct:
```c++
struct HitRootEvent
{
    int event_id{0};
    std::map<std::string, std::vector<G4VHit*>> hcmap;
};
```
since it basically makes all the member data completely accessible.

Reply:
Okay, changed to a simple struct as suggested.